DatadogRUM/Tests/RUMMonitor/Scopes/RUMSessionScopeTests.swift

```diff
@@ -361,13 +363,13 @@ class RUMSessionScopeTests: XCTestCase {
             hasTrackedAnyView: false,
             didStartWithReplay: randomIsReplayBeingRecorded
         )
-        let baggageSent = try XCTUnwrap(featureScope.messagesSent().lastBaggage(withKey: RUMBaggageKeys.sessionState))
-        let sessionStateSent: RUMSessionState = try baggageSent.decode()
-        XCTAssertEqual(sessionStateSent, expectedSessionState, "It must send 'session state' message")
+        let actualSessionState = try XCTUnwrap(fatalErrorContext.sessionState)
+        XCTAssertEqual(actualSessionState, expectedSessionState)
```

Comment:
💡 This change in `RUMSessionScopeTests` is around asserting the update on `FatalErrorContext` instead of inspecting the baggage sent over message bus. The baggage is now tested in `FatalErrorContextTests` which streamlines things and better separates concerns.